### DISCUSSION TITLE
Nested Drawer Logic

### DISCUSSION THREAD

**Discussion by noobCode-69** [vercel/next.js#63098] [App Router]
Hey,

I need help implementing a feature: I want a modal to open in a separate route, like "/mypage/drawer1", when a button is clicked on "/mypage". I tried using nested routes (app directory) , The only issue is when i close the modal and   click the browser back button, the modal reopens (navigate back to 'mypage/drawer1') and I don't want that.
 Any quick fixes or am i doing it complete wrong

Thanks PS:)

@samselikoff 
[www_screencapture_com_2024-3-10_04_32.webm](https://github.com/vercel/next.js/assets/76926762/3f662e1b-e177-4396-9092-b2d460a5697f)

**Comment by samselikoff** [comment]
I actually have a YouTube video showing refactoring this very UX – give it a watch and let me know if it answers your questions!

https://youtu.be/-UskvNS13yw?si=tAshOalRndIVAI1e

**Reply by noobCode-69** [reply]
Thanks for the amazing video , it is exactly what I wanted . But would it be possible that If i click back I don't reopen my drawer.
